src/WebJobs.Script/Host/ScriptHost.cs

```diff
@@ -1131,6 +1134,74 @@ public static Collection<FunctionMetadata> ReadFunctionMetadata(ScriptHostConfig
             return functions;
         }
 
+        public static Collection<FunctionMetadata> ReadProxyMetadata(ScriptHostConfiguration config, out object proxyClient, IProxyClientGenerator proxyClientGenerator = null, ScriptSettingsManager settingsManager = null)
+        {
+            proxyClient = null;
+
+            // read the proxy config
+            string proxyConfigPath = Path.Combine(config.RootScriptPath, ScriptConstants.ProxyMetadataFileName);
+            if (!File.Exists(proxyConfigPath))
+            {
+                return null;
+            }
+
+            settingsManager = settingsManager ?? ScriptSettingsManager.Instance;
+
+            var proxyAppSettingValue = settingsManager.GetSetting(EnvironmentSettingNames.ProxySiteExtensionEnabledKey);
+
+            // This is for backward compatibility only, if the file is present but the value of proxy app setting(ROUTING_EXTENSION_VERSION) is explicitly set to 'disabled' we will ignore loading the proxies.
+            if (!string.IsNullOrWhiteSpace(proxyAppSettingValue) && proxyAppSettingValue.Equals("disabled", StringComparison.OrdinalIgnoreCase))
+            {
+                return null;
+            }
+
+            var proxies = new Collection<FunctionMetadata>();
+            string proxiesJson = File.ReadAllText(proxyConfigPath);
+
+            LoadProxyRoutes(config, proxies, out proxyClient, proxiesJson, proxyClientGenerator);
+
+            return proxies;
+        }
+
+        [System.Runtime.CompilerServices.MethodImpl(System.Runtime.CompilerServices.MethodImplOptions.NoInlining)]
```

Comment:
This is so that the complier dosnt make this function an inline during optimization. This ensures that the Shim dll will not be loaded unless it is used.

---
In reply to: [132542352](https://github.com/Azure/azure-webjobs-sdk-script/pull/1749#discussion_r132542352) [](ancestors = 132542352)